### PR TITLE
Milestone Backend: Rails Form

### PR DESCRIPTION
**Pull request by pointerish**
This PR implements the following:

- A `user_controller`
- Views for `edit`, `new`
- `User` model
- Rails form as interface

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  def edit
+    @user = User.find(params[:id])
+  end
+
+  def update
+    @user = User.find(params[:id])
+    @user.update(user_params)
+    render :edit if @user.save
+  end
```
Comment (teekaytech) [inline app/controllers/users_controller.rb:24]: For your update action, 
`redirect_to edit_user_path` if the user was saved
`render :edit` if the user was not saved ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_update.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've done an excellent job in this project.
However, some changes are required before it can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [ ] Remove the `node_modules` directory from your repo. This directory should not be committed to your repository. Kindly delete it, add it to your `.gitignore` file, and commit your changes.

- [ ] The commented out HTML `<form>` is missing. You are expected to leave the commented `<form>` tag and not delete it totally. Kindly do so. ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_tag2.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))

- [ ] Also, you should leave the commented out `form_tag` and not delete it totally.
[Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_form_for.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms)


**[ OPTIONAL CHANGES ]**

* [ ] You can create a partial for user form and reuse it in edit and new views
* [ ] You can also make your form view to display a list of the error messages (with #errors and full_messages usage) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/forms/user_extra_shared_error.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/forms))
* [ ] validate the email format by using either a regular expression or a built-in URI method. [This stack overflow thread](https://stackoverflow.com/questions/38611405/email-validation-in-ruby-on-rails) might help.

* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Comment by pointerish** [comment]
Thank you for your thouroughness, Taofeek. We'll work on the improvements.

**Comment by pointerish** [comment]
The changes were made. Thanks, @teekaytech for pointing out the improvements. It was interesting to implement them.

In order to comment out the previous versions of the form I added a helper empty method to `application_helper.rb` called `comment`. That way I can simply wrap the unneeded form `erb` code inside `<% comment do %>` and `<% end %>` and Ruby will yield the wrapped template into oblivion. A very nice trick found on this StackOverflow [post](https://stackoverflow.com/questions/3127644/block-comments-in-html-erb-templates-in-rails).

The validation improvement was interesting as well as the feature where the errors on the form are rendered inside `<li>` tags.

Regards,
Josias Alvarado

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

The way of commenting is something I have never seen before. Keep up this out-of-the-box thinking!

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Comment by pointerish** [comment]
Hi @tirthajyoti-ghosh, thanks for the nice remarks and for the help provided! Happy coding.

**Pull request merged**